test/e2e/tests/ppom-blockaid-alert-metrics.spec.js

```diff
@@ -0,0 +1,185 @@
+            //   type: "eth_signTypedData",
+            //   ui_customizations: [
+            //     "flagged_as_malicious"
+            //   ]
+            // },
+          },
+        ],
+      })
+      .thenCallback(() => {
+        return {
+          statusCode: 200,
+        };
+      }),
+  ];
+}
+
+describe('Confirmation Security Alert - Blockaid @no-mmi', function () {
+  it('should show security alerts for malicious requests', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withNetworkControllerOnMainnet()
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesController({
+            securityAlertsEnabled: true,
+          })
+          .withMetaMetricsController({
+            metaMetricsId: 'fake-metrics-id',
+            participateInMetaMetrics: true,
+          })
+          .build(),
+        defaultGanacheOptions,
+        testSpecificMock: mockInfuraWithMaliciousResponses,
+        title: this.test.fullTitle(),
+      },
+
+      async ({ driver, mockedEndpoint: mockedEndpoints }) => {
+        await driver.navigate();
+        await unlockWallet(driver);
+        await openDapp(driver);
+
+        for (const config of testMaliciousConfigs) {
+          const { expectedDescription, expectedReason, btnSelector } = config;
+
+          // Click TestDapp button to send JSON-RPC request
+          await driver.clickElement(btnSelector);
+
+          // Wait for confirmation pop-up
+          const windowHandles = await driver.waitUntilXWindowHandles(3);
+          await driver.switchToWindowWithTitle(
+            WINDOW_TITLES.Notification,
+            windowHandles,
+          );
+
+          // Find element by title
+          const bannerAlertFoundByTitle = await driver.findElement({
+            css: bannerAlertSelector,
+            text: expectedMaliciousTitle,
+          });
+          const bannerAlertText = await bannerAlertFoundByTitle.getText();
+
+          assert(
+            bannerAlertFoundByTitle,
+            `Banner alert not found. Expected Title: ${expectedMaliciousTitle} \nExpected reason: ${expectedReason}\n`,
+          );
+          assert(
+            bannerAlertText.includes(expectedDescription),
+            `Unexpected banner alert description. Expected: ${expectedDescription} \nExpected reason: ${expectedReason}\n`,
+          );
+
+          await driver.clickElement({ text: 'See details', tag: 'p' });
+
+          await driver.clickElement({ text: 'Contact us', tag: 'a' });
+
+          // Wait for confirmation pop-up to close
+          await driver.clickElement({ text: 'Reject', tag: 'button' });
+          await driver.switchToWindowWithTitle(
+            WINDOW_TITLES.TestDApp,
+            windowHandles,
+          );
+
+          console.log('mockedEndpoints', mockedEndpoints);
+          const events = await getEventPayloads(driver, mockedEndpoints);
```

Comment:
@seaona I'm having a bit of challenge here. This call times out. It works in the other file, but fails here.

I checked when I opened the alert from testdapp manually that metrics is indeed actually sent.

Reply:
So the problem is this: the event I was looking for is not being sent during e2e, but it's being sent normally when I check the network tab.